### PR TITLE
load & set default currency

### PR DESCRIPTION
**Pull request by iurimatias**
not fully done yet, to avoid colliding with other wallet work

### CODE REVIEW

File: src/app/wallet/view.nim [hunk 1]
```diff
@@ -39,3 +39,19 @@ QtObject:
 
   proc getDefaultAccount*(self: WalletView): string {.slot.} =
     return self.defaultAccount
+
+  proc defaultCurrency*(self: WalletView): string {.slot.} =
+    # self.defaultCurrency
```
Comment (jrainville) [inline src/app/wallet/view.nim:44]: Can remove

File: src/app/wallet/view.nim [hunk 2]
```diff
@@ -39,3 +39,19 @@ QtObject:
...
+    self.status.wallet.getDefaultCurrency()
+
+  proc defaultCurrencyChanged*(self: WalletView) {.signal.}
+
+  proc setDefaultCurrency*(self: WalletView, currency: string) {.slot.} =
+    # self.defaultCurrency = channel
```
Comment (jrainville) [inline src/app/wallet/view.nim:50]: Same hre

File: src/status/libstatus/settings.nim [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import core
+import json
+
+# proc saveSettings*(key: string, value: JsonNode): string =
```
Comment (jrainville) [inline src/status/libstatus/settings.nim:4]: Can remove

File: src/status/libstatus/core.nim [hunk 4]
```diff
@@ -16,7 +16,13 @@ proc callPrivateRPC*(methodName: string, payload = %* []): string =
...
       "params": %payload
     }
-    result = $libstatus.callPrivateRPC($inputJSON)
+    echo "calling json"
+    echo $inputJSON
+    let response = libstatus.callPrivateRPC($inputJSON)
+    result = $response
+    if parseJSON(result).hasKey("error"):
+      echo "rpc response error"
+      echo result
```
Comment (jrainville) [inline src/status/libstatus/core.nim:25]: Do you still need the echos? If so, it's better to use `debug` or another chronicles method

File: src/status/libstatus/wallet.nim [hunk 5]
```diff
@@ -28,8 +28,11 @@ proc getPrice*(crypto: string, fiat: string): string =
   let client = newHttpClient()
   client.headers = newHttpHeaders({ "Content-Type": "application/json" })
 
-  let response = client.request(url)
-  $parseJson(response.body)["USD"]
+  try:
+    let response = client.request(url)
+    result = $parseJson(response.body)[fiat.toUpper]
+  except:
+    echo "error getting price"
```
Comment (jrainville) [inline src/status/libstatus/wallet.nim:35]: Use `error` from chronicles

File: ui/app/AppLayouts/Wallet/SettingsModal.qml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    function open() {
+        popup.open()
+        settingsModalContent.currency = assetsModel.defaultCurrency
+    }
+
+    function close() {
+        popup.close()
+    }
+
+    Popup {
```
Comment (jrainville) [inline ui/app/AppLayouts/Wallet/SettingsModal.qml:19]: Richard created a reusable component for modal popups that simplifies the process heavily. It's better to use it.

Comment (iurimatias) [reply ui/app/AppLayouts/Wallet/SettingsModal.qml:19]: we'll do this later together with the send modal

**Pull request merged**